### PR TITLE
machine: list/create/delete/rollback snapshots

### PR DESCRIPTION
**Pull request by jvwilder**
#### What this PR resolves:

#76: create/delete/list/rollback snapshots + docs/tests

### CODE REVIEW

**Comment by jvwilder** [comment]
Requires Jumpscale/lib9#48

File: templates/nodes/node.ovc/actions.py [hunk 1]
```diff
@@ -330,7 +330,7 @@ def processChange(job):
 
                 setattr(service.model.data, key, value)
 
-            if key == 'cloneName':
+            if key == 'cloneName' or  key == 'snapshotEpoch':
```
Comment (xmonader) [inline templates/nodes/node.ovc/actions.py]: maybe you can use `key in ['cloneName', 'snapshotEpcoh']`

File: tests/test_services/test_node_ovc/actions.py [hunk 2]
```diff
@@ -216,3 +222,92 @@ def test_clone(job):
...
+            service.model.data.result = RESULT_OK % 'test_snapshot'
+
+    except Exception as e:
+        service.model.data.result = RESULT_ERROR % (str(sys.exc_info()[:2]) + str(e))
+    service.save()
+
+
+def test_list_snapshots(job):
+    import sys
+    import json
```
Comment (xmonader) [inline tests/test_services/test_node_ovc/actions.py]: Why not use j.data.serializers.json?

File: tests/test_services/test_node_ovc/actions.py [hunk 3]
```diff
@@ -216,3 +222,92 @@ def test_clone(job):
...
+
+        vm = service.producers['node'][0]
+        vm_id = vm.model.data.machineId
+
+        snapshots = client.api.cloudapi.machines.listSnapshots(machineId=vm_id)
+
+        actual_snapshots = [json.loads(s) for s in vm.model.data.snapshots]
+
+        # check if snapshot lists match
+        if snapshots != actual_snapshots:
```
Comment (xmonader) [inline tests/test_services/test_node_ovc/actions.py]: What ensures the order here? maybe you can compare against a `set` of those objects or sort both lists?

**Pull request merged**